lib/router/mbaas-auth.js

```diff
@@ -0,0 +1,29 @@
+const q = require('q');
+
+/**
+ * @param {Object} mediator object used to subscribe and publish read and auth topics
+ * @param {String} username Unique identifier of the user.
+ * @param {String} password passed in password to be verified against user.
+ *
+ */
+exports.auth = function(mediator, username, password) {
+
+  function checkAuthentication(authenticated) {
+    var checkAuthDeferred = q.defer();
+    if (authenticated) {
+      console.log(username, ' authenticated');
+      return mediator.request('wfm:user:username:read', username);
```

Comment:
If you are returning the promise from `mediator.request('wfm:user:username:read', username);`, the `checkAuthDeferred` promise will never have been resolved. It may cause a memory leak.

I'd say only create the `checkAuthDeferred` if it is not authenticated.